src/banded.jl

```diff
@@ -0,0 +1,550 @@
+ 11  22  33  44  55  65  77   0
+ 21  32  43  54  64  76   0   0
+ 31  42  53  63  75   0   0   0
+ 41  52   0  74   0   0   0   0
+```
+"""
+function batchedbandedlu!(
+        mat::A,
+        kl = div(size(mat, 2) - 1, 2)
+    ) where {T, A <: AbstractArray{T, 4}}
+    (Nqh, width, n, Neh) = size(mat)
+    ku = width - 1 - kl
+    fac = BatchedBandedLU{Nqh, n, ku, kl, Neh, T, A}(mat)
+    return batchedbandedlu!(fac)
+end
+
+function batchedbandedlu!(
+        fac::BatchedBandedLU{Nqh, n, ku, kl, Neh, T, A}
+    ) where {Nqh, n, ku, kl, Neh, T, A}
+    event = Event(device(A))
+    kernel! = bandedlu_kernel!(device(A), (Nqh,))
+    event = kernel!(fac; ndrange = (Nqh * Neh,), dependencies = (event,))
+    wait(event)
+    return fac
+end
+
+@kernel function bandedlu_kernel!(
+        fac::BatchedBandedLU{Nqh, n, ku, kl, Neh}
+    ) where{Nqh, n, ku, kl, Neh}
+
+    @uniform begin
+        # center index of band
+        c = ku + 1
+    end
+
+    # horizonal element number
+    eh = @index(Group, Linear)
+
+    # horizontal degree of freedom
+    ij = @index(Local, Linear)
+
+    # Create an object that indexes like a matrix for this thread
+    U = L = view(fac, ij, :, :, eh)
+
+    # matrix index: (u,v) -> banded index: (c + u - v, u)
+    # v is column index
+    @inbounds for v = 1:n
+        # Get the pivot
+        # Uvv = U[c, v]
+        Uvv = U[v, v]
+
+        # Fill L
+        for p = 1:kl
+            # L[v + p, v] = U[v + p, v] / U[v, v]
+            # L[c + p, v] /= Uvv
+            L[v + p, v] /= Uvv
+        end
+
+        # Update U
+        for q = 1:ku
+            # u is row index
+            u = v + q
+            # If this row is part of the matrix update it
+            if u ≤ n
+                # Uvu = U[c - q, u]
+                Uvu = U[v, u]
+                for p = 1:kl
+                    # U[v + p, u] -= L[v + p, v] * U[v, u]
+                    # U[c + p - q, u] -= L[c + p, v] * Uvu
+                    U[v + p, u] -= L[v + p, v] * Uvu
+                end
+            end
+        end
+    end
+end
+
+@kernel function banded_forward_kernel!(
+        x_::AbstractArray{T, 3},
+        fac::BatchedBandedLU{Nqh, n, ku, kl, Neh, T},
+        b_::AbstractArray{T, 3},
+    ) where{Nqh, n, ku, kl, Neh, T}
+
+    # private storage for the part of b we are working on
+    p_b = @private T (kl + 1)
+
+    # horizonal element number
+    eh = @index(Group, Linear)
+
+    # horizontal degree of freedom
+    ij = @index(Local, Linear)
+
+    # Create an object that indexes like a matrix for this thread
+    L = view(fac, ij, :, :, eh)
+    b = view(b_, ij, :, eh)
+    x = view(x_, ij, :, eh)
+
+    # Fill the private storage of b
+    @inbounds for v = 1:kl+1
+        p_b[v] = v ≤ n ? b[v] : -zero(T)
+    end
+
+    # Loop over the columns
+    @inbounds for v = 1:n
+        # Pull out the b associated with v
+        x[v] = bv = p_b[1]
+
+        # Loop over the rows
+        for p = 1:kl
```

Comment:
```suggestion
        @unroll for p = 1:kl
```
Again, not sure if this is an improvement.